core/src/test/java/com/linecorp/armeria/common/AggregatedHttpObjectTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Unit tests for {@link AggregatedHttpObject}.
+ * @author Jinwoo Bae
+ */
+public class AggregatedHttpObjectTest {
+
+    @Test
+    void testDecodingContentCorrectly() {
+        final HttpHeaders headers = HttpHeaders.of(HttpHeaderNames.CONTENT_TYPE, "text/plain; charset=utf-8");
+        final HttpData httpData = HttpData.of(StandardCharsets.UTF_8, "foo");
+
+        final AggregatedHttpObject aggregatedHttpObject = new MockAggregateHttpObject(headers, httpData);
```

Comment:
What do you think of just using public APIs instead of mocking?

i.e.
```
final RequestHeaders headers = RequestHeaders.of(
        HttpMethod.GET, "/",
        HttpHeaderNames.CONTENT_TYPE, "text/plain; charset=utf-8");
final HttpData httpData = HttpData.of(StandardCharsets.UTF_8, "foo");
final AggregatedHttpRequest request =
        HttpRequest.of(headers, httpData).aggregate().join();
```

Reply:
That seems simpler. I'll change it.